kolibri/plugins/coach/assets/src/app.js

```diff
@@ -38,18 +39,34 @@ class CoachToolsModule extends KolibriApp {
       this.store.commit('SET_PAGE_NAME', to.name);
       if (
         to.name &&
+        !to.params.classId &&
         !['CoachClassListPage', 'StatusTestPage', 'CoachPrompts', 'AllFacilitiesPage'].includes(
           to.name
         )
       ) {
-        if (to.params.classId)
-          promises.push(this.store.dispatch('initClassInfo', to.params.classId));
-      } else {
         this.store.dispatch('coachNotifications/stopPolling');
       }
+      // temporary condition as we're gradually moving all promises below this line to local page handlers and therefore need to skip those that we already refactored here https://github.com/learningequality/kolibri/issues/11219
+      if (to.name && to.name === PageNames.EXAMS) {
+        next();
```

Comment:
I believe we also need to add `return;` below `next()` to ensure that code execution is stopped